File: ios-app/InvestmentTracker/Models/Portfolio/PortfolioAnalytics.swift
import Foundation

struct PortfolioHistoryPoint: Codable, Identifiable {
    let id = UUID()
    let date: Date
    let value: Double
    let change: Double
    let changePercent: Double
    
    private enum CodingKeys: String, CodingKey {
        case date, value, change, changePercent
    }
}

struct AssetAllocationPoint: Codable, Identifiable {
    let id = UUID()
    let assetType: String
    let assetName: String
    let value: Double
    let percentage: Double
    let color: String
    
    private enum CodingKeys: String, CodingKey {
        case assetType, assetName, value, percentage, color
    }
}

struct TopMover: Codable, Identifiable {
    let id = UUID()
    let assetId: String
    let assetSymbol: String
    let assetName: String
    let currentPrice: Double
    let change: Double
    let changePercent: Double
    let value: Double
    let direction: String
    
    private enum CodingKeys: String, CodingKey {
        case assetId, assetSymbol, assetName, currentPrice, change, changePercent, value, direction
    }
}

struct PortfolioAnalytics: Codable {
    let portfolioHistory: [PortfolioHistoryPoint]
    let assetAllocation: [AssetAllocationPoint]
    let topMovers: [TopMover]
    let totalReturn: Double?
    let totalReturnPercent: Double?
    let volatility: Double?
    let sharpeRatio: Double?
    let maxDrawdown: Double?
}

File: android-app/app/src/main/java/com/yuksel/investmenttracker/data/model/analytics/PortfolioAnalytics.kt
package com.yuksel.investmenttracker.data.model.analytics

import com.google.gson.annotations.SerializedName
import java.time.LocalDate

data class PortfolioHistoryPoint(
    @SerializedName("date") val date: LocalDate,
    @SerializedName("value") val value: Double,
    @SerializedName("change") val change: Double,
    @SerializedName("changePercent") val changePercent: Double
)

data class AssetAllocationPoint(
    @SerializedName("assetType") val assetType: String,
    @SerializedName("assetName") val assetName: String,
    @SerializedName("value") val value: Double,
    @SerializedName("percentage") val percentage: Double,
    @SerializedName("color") val color: String
)

data class TopMover(
    @SerializedName("assetId") val assetId: String,
    @SerializedName("assetSymbol") val assetSymbol: String,
    @SerializedName("assetName") val assetName: String,
    @SerializedName("currentPrice") val currentPrice: Double,
    @SerializedName("change") val change: Double,
    @SerializedName("changePercent") val changePercent: Double,
    @SerializedName("value") val value: Double,
    @SerializedName("direction") val direction: String
)

data class PortfolioAnalytics(
    @SerializedName("portfolioHistory") val portfolioHistory: List<PortfolioHistoryPoint>,
    @SerializedName("assetAllocation") val assetAllocation: List<AssetAllocationPoint>,
    @SerializedName("topMovers") val topMovers: List<TopMover>,
    @SerializedName("totalReturn") val totalReturn: Double?,
    @SerializedName("totalReturnPercent") val totalReturnPercent: Double?,
    @SerializedName("volatility") val volatility: Double?,
    @SerializedName("sharpeRatio") val sharpeRatio: Double?,
    @SerializedName("maxDrawdown") val maxDrawdown: Double?
)